### DISCUSSION TITLE
Programatically import code to github

### DISCUSSION THREAD

**Discussion by Vkl279** [community/community#41104] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi Team,
I want to programmatically create repo and import code to github. Please suggest suitable options

**Comment by airtower-luna** [comment]
You can create repositories [using the Repositories API](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28). You can make commits using the API, too (see [Contents API](https://docs.github.com/en/rest/repos/contents?apiVersion=2022-11-28)), but if you have more than a single file or two it's probably better to find a Git library for your programming language and use proper Git operations.

**Reply by Vkl279** [reply]
Thanks. Could you please advise related to Java, any git library that I can use?

**Reply by Vkl279** [reply]
If I use a git library,  is it possible to create a repo?

**Reply by airtower-luna** [reply]
> Could you please advise related to Java, any git library that I can use?

I know Eclipse uses [JGit](https://www.eclipse.org/jgit/) for its Git integration. I haven't used it myself beyond using Eclipse, though.

> If I use a git library, is it possible to create a repo?

Locally, yes. To create a repository on GitHub you'll need to use the API. Practically you'll probably have to do both (assuming you want a completely new repo, not edit an existing one): Create the repository locally and commit the files, create the repository on GitHub, add remote and push.

**Reply by Vkl279** [reply]
In the steps you have mentioned,  only "create repository on github" requires API usage right? Is it the curl option? I have also come across github command-line, gh and ghub
Once repo is created on github, i can then add it as remote to the local repo and push?

**Reply by airtower-luna** [reply]
> In the steps you have mentioned, only "create repository on github" requires API usage right?

Yes. The API is accessed over HTTPS. If you send the request with `curl`, using `gh`, or some other tool doesn't matter, but if you're developing in Java I suggest making the HTTPS request from Java, too, instead of calling some other tool. :wink:
